### PR TITLE
rpc: Use the return value of GetProxy(...) in GetNetworksInfo(). Mark GetProxy(...) with [[nodiscard]].

### PR DESCRIPTION
**Pull request by practicalswift**
* Use the return value of `GetProxy(...)` in `GetNetworksInfo()`.
* Mark `GetProxy(...)` with `[[nodiscard]]`.

### CODE REVIEW

File: src/rpc/net.cpp [hunk 1]
```diff
@@ -419,14 +419,17 @@ static UniValue GetNetworksInfo()
...
         UniValue obj(UniValue::VOBJ);
-        GetProxy(network, proxy);
         obj.pushKV("name", GetNetworkName(network));
         obj.pushKV("limited", !IsReachable(network));
         obj.pushKV("reachable", IsReachable(network));
-        obj.pushKV("proxy", proxy.IsValid() ? proxy.proxy.ToStringIPPort() : std::string());
-        obj.pushKV("proxy_randomize_credentials", proxy.randomize_credentials);
+
+        proxyType proxy;
+        bool useProxy = GetProxy(network, proxy);
```
Comment (promag) [inline src/rpc/net.cpp]: nit, use_proxy.

**Review by promag — COMMENTED** [review]
utACK

**Comment by practicalswift** [comment]
@promag Thanks for reviewing! Feedback addressed. Please re-review :-)

File: src/netbase.h [hunk 2]
```diff
@@ -40,7 +41,7 @@ class proxyType
 enum Network ParseNetwork(std::string net);
 std::string GetNetworkName(enum Network net);
 bool SetProxy(enum Network net, const proxyType &addrProxy);
-bool GetProxy(enum Network net, proxyType &proxyInfoOut);
+NODISCARD bool GetProxy(enum Network net, proxyType &proxyInfoOut);
```
[the remark this answers is not in the document]
Comment (laanwj) [reply src/netbase.h]: AHHH this gets into trouble in `OptionsDialog::updateDefaultProxyNets` which seems to rely on the struct *even if* the return value and proxy isn't valid.

**Comment by laanwj** [comment]
Here's my branch that uses `Optional`: https://github.com/laanwj/bitcoin/tree/2019_02_optional

The only thing I'm not sure about is whether the options dialog change is handled correctly, what the proxy string should be in case of an nonexistent/invalid proxy.

**Comment by practicalswift** [comment]
@laanwj Your solution looks better. I'm closing this PR: please consider submitting your PR instead.

**Comment by laanwj** [comment]
Thanks!
I'll try to get back to it after 0.18.

**Pull request closed without merging**